### ISSUE TITLE
Add JSON Schema for more universal spec

### ISSUE THREAD

**Issue by SelfhostedPro** [code4recovery/spec#105]
Hey, currently the way the spec is defined, it makes it difficult to work with automated tooling outside of PHP.

It would be a very nice plus if there was a JSON Schema as it's widely used and there are lots of tools for using it:
Docs: https://json-schema.org/overview/what-is-jsonschema
Tools: https://json-schema.org/tools

I would be happy to help if needed, just let me know. I am really appreciative that there are any docs at all on this.

**Comment by SelfhostedPro** [comment]
I was also looking and don’t see anything about which type of program the meeting belongs to in the spec. (ie. AA, NA, CA, etc. ) I definitely understand the primary focus of this being AA but was wondering if there are any proposals floating around for a field like that so it could be utilized by other 12 step programs?

My main use case is building a website for an Alano Club (ideally a reusable nuxt template any club can use).

If you would like to communicate about this elsewhere please let me know as I have a discord for my other projects and am working on joining the AA tech forum.

**Comment by SelfhostedPro** [comment]
Yeah, it essentially would allow specifying it once in this format as a source of truth and then generating from that wherever else you’d want to use it. (Kind of like an OpenAI spec if you’ve used them before)
(ie. For PHP: https://dev.to/robertobutti/validating-json-with-json-schema-and-php-2b4i)

It would likely fit best here. 

I don’t think it’s super necessary to add a program key yet, just makes my one specific use case easier. I definitely understand not wanting to step on their toes about it. 

I’ll shoot an email over shortly. Thanks for getting back to me!

In the meantime, is there a specific file that has a full overview of the current spec that’s typed or should I piece it together from the readme?

**Comment by joshreisner** [comment]
the readme section called Field Definitions is a complete definition, so it should have what you need
